airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/pipline/object_storage/ObjectLoaderPipeline.kt

```diff
@@ -27,10 +31,36 @@ import jakarta.inject.Singleton
  * - a single completer worker reads the second queue and completes the uploads
  * - state is acked only when the completer finishes each upload
  */
-@Singleton
-@Requires(bean = ObjectLoader::class)
+//@Singleton
+//@Requires(bean = ObjectLoader::class)
+@Requires(property = "airbyte.destination.core.file-transfer.enabled", value = "false")
 class ObjectLoaderPipeline<K : WithStream, T : RemoteObject<*>>(
     partStep: ObjectLoaderPartFormatterStep,
-    uploadStep: ObjectLoaderPartLoaderStep<T>,
-    completerStep: ObjectLoaderUploadCompleterStep<K, T>,
+    @Named("recordPartLoaderStep") uploadStep: ObjectLoaderPartLoaderStep<T>,
+    @Named("recordUploadCompleterStep") completerStep: ObjectLoaderUploadCompleterStep<K, T>,
 ) : LoadPipeline(listOf(partStep, uploadStep, completerStep))
+
+
+
+@Singleton
+@Requires(bean = ObjectLoader::class)
+//@Requires(property = "airbyte.destination.core.file-transfer.enabled", value = "true")
```

Comment:
Todo: We need to replace this pipeline selection logic to only create the ObjectLoaderPipelineWithFiles if files are gonna move.